### PR TITLE
feature: independent runtime root

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### Ⅰ. Describe what this PR did

put different runtime to independent runtime root.

First, distinguish different runtime files. Secondly, it enables v1 containerd-shim to know which runtime to go (hack).

### CODE REVIEW

**Comment by allencloud** [comment]
Sounds good!

File: ctrd/container.go [hunk 1]
```diff
@@ -540,7 +541,7 @@ func (c *Client) createContainer(ctx context.Context, ref, id, checkpointDir str
 		containerd.WithContainerLabels(container.Labels),
 		containerd.WithRuntime(fmt.Sprintf("io.containerd.runtime.v1.%s", runtime.GOOS), &runctypes.RuncOptions{
 			Runtime:       container.Runtime,
-			RuntimeRoot:   runtimeRoot,
+			RuntimeRoot:   path.Join(runtimeRoot, container.Runtime),
```
Comment (allencloud) [inline ctrd/container.go]: Do we first create this directory or someone would create it if non-exist?
Just confirm.

Comment (zhuangqh) [reply ctrd/container.go]: > Do we first create this directory or someone would create it if non-exist?
> Just confirm.

The existence of the path and the security of the join are promised by runtime. 😄

**Comment by allencloud** [comment]
LGTM

**Comment by Ace-Tang** [comment]
We still should check if it is compatible for exist container

**Comment by Ace-Tang** [comment]
I check the containerd shim will keep the runtime root, so it should not effect exist container
```
// Config contains shim specific configuration
type Config struct {
    Path          string
    Namespace     string
    WorkDir       string
    Criu          string
    RuntimeRoot   string
    SystemdCgroup bool
}
```

**Pull request merged**